src/OrchardCoreContrib.PoExtractor.DotNet.CS/MetadataProviders/CSharpMetadataProvider.cs

```diff
@@ -29,8 +30,14 @@ public string GetContext(SyntaxNode node)
                 .Name.ToString();
             var @class = node.Ancestors()
                 .OfType<ClassDeclarationSyntax>()
-                .FirstOrDefault()?
-                .Identifier.ValueText;
+                .Select(x => x.Identifier.ValueText)
```

Comment:
Just few comments:

- We should do this for both C# & VB
- Adding a unit test will be a plus

Reply:
sorry, i am not family with VB. feel free to close this PR and add another